src/directives/profile.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @fileoverview Provides a directive used to insert an elevation profile chart
+ * in the DOM.
+ *
+ * Example:
+ *
+ * <div ngeo-profile="ctrl.profileData"
+ *      ngeo-profile-onhover="ctrl.onHover(point)"
+ *      ngeo-profile-onout="ctrl.onProfileOut()"></div>
+ */
+goog.provide('ngeo.profileDirective');
+
+goog.require('ngeo');
+goog.require('ngeo.profile');
+
+
+/**
+ * @return {angular.Directive} Directive Definition Object.
+ */
+ngeo.profileDirective = function() {
+  return {
+    restrict: 'A',
+    link:
+        /**
+         * @param {angular.Scope} scope Scope.
+         * @param {angular.JQLite} element Element.
+         * @param {angular.Attributes} attrs Attributes.
+         */
+        function(scope, element, attrs) {
+
+          var selection = d3.select(element[0]);
+          var profile = ngeo.profile();
+
+          scope.$watch(attrs['ngeoProfile'], function(newVal, oldVal) {
+            if (goog.isDef(newVal)) {
+              selection.datum(newVal['profile']).call(profile);
+            }
+          });
+
+          profile.onHover(function(point) {
+            scope.$eval(attrs['ngeoProfileOnhover'], {'point': point});
+          });
+
+          profile.onOut(function() {
+            scope.$eval(attrs['ngeoProfileOnout']);
+          });
```

Comment:
I plan to introduce an `options` to pass to the `ngeo.profile` factory.
It would look like:
```js
/**
 * Options for the D3 Profile.
 *  @typedef {{
 *    light: (boolean|undefined),
 *    extractor: ngeox.profile.ProfileExtractor,
 *    ... other options including hover and out
 *  }}
 */
ngeox.profile.ProfileOptions;
```

@elemoine, @pgiraud, is the next step to add a `ngeoProfileOptions` attribute and remove the `ngeoProfileOnhover` and `ngeoProfileOnout` attributes? Then in the directive to call `scope.$watchCollection(attrs['ngeoProfileOptions'],` and recreate the profile on change?